docs/understand/kv-in-a-nutshell.md

```diff
@@ -0,0 +1,51 @@
+  * `Commit` A finishes, tries to update the `Branch` and fails due to the recent modification by `Commit` B - the `StagingToken` is set to tokenB and not tokenA as expected by `Commit` A,
+  * `Commit` B finishes and updates the branch, as tokenB is set as `StagingToken` and it matches the flow expectation
+
+An important detail to note, is that as a `Commit` starts, and the `StagingToken` is set a new value, the former value is added to a list of 'still valid' `StagingToken`s - `SealedToken` - on the `Branch`, which makes sure no `StagingToken` and no object are lost due to a failed `Commit`
+
+You can read more on the Commit Flow in the [dedicated section in the KV Design](https://github.com/treeverse/lakeFS/blob/master/design/accepted/metadata_kv/index.md#graveler-metadata---branches-and-staged-writes)
+
+## DB Transactions and Atomic Updates
+
+Another notable difference is the existence of DB transactions with PostgreSQL, ability that our KV Store lacks. This ability was leveraged by `lakeFS` to construct several DB updates, into one "atomic" operation - each failure, in each step, rolled back the entire operation, keeping the DB consistent and clean.
+With KV Store, this ability is gone, and we had to come up with various solutions. As a starting point, the DB consistency is, obviously, not anything we can risk. On the other hand, maintaining the DB clean, and as a result smaller, is something that can be sacrificed, at least as a first step. Let us take a look at a relatively simple flow of a new `Repository` creation:
+A brand new `Repository` has 3 objects: The `Repository` object itself, an initial `Branch` object and an initial `Commit`, which the `Branch` points to. With SQL DB, it was as simple as creating all 3 objects in the DB under one transaction (at this order). Any failure resulted in a rollback and no redundant leftovers in our DB.
+With no transaction in KV Store, if for example the `Branch` creation fails, it will leave the `Repository` without an initial `Branch` (or a `Branch` at all), yet the `Repository` will be accessible. Trying to delete the `Repository` as a response to `Branch` creation failure is ony a partial solution as this operation can fail as well.
+To mitigate this we introduced a per-`Repository`-partition, which holds all repository related objects (the `Branch` and `Commit` in this scenario). The partition key can only be derived from the specific`Repository` instance itself. In addition we first create the `Repository` objects, the `Commit` and the `Branch`, under the `Repository`'s partition key, and then the `Repository` is created. The `Repository` and its objects will be accessible only after a successful creation of all 3 entities. A failure in this flow might leave some dangling objects, but consistency is maintained.
+The amount of such dangling objects is not expected to be significant, and we plan to implement a cleaning algorithm to keep our KV Store neat and clean
```

Comment:
Maybe add a summary at the end - and finish with a more positive note 😄

Reply:
I'm not sure I agree with this approach. This is a technical page and I tried to describe design challenges we were facing, and the solutions we came up with. I don't think it needs to end "happily", as this doc is not coming to convivce anyone of anything, IMO - it comes to provide an explanation
WDYT?

Reply:
I'll try to rephrase:
This document contains a lot of technical information and is very verbose. At the end of the document - the reader might be confused as to what exactly are the changes that were described in the document.
So I feel it would benefit the reader to get a concise summary of the changes in the form or bullets or otherwise, at the beginning of the document or at the end

Reply:
Added